qa/common/gen_qa_models.py

```diff
@@ -1280,6 +1280,229 @@ def create_onnx_modelconfig(models_dir, max_batch, model_version, input_shape,
+                            output0_dtype,
+                            output1_dtype,
+                            swap=False):
+
+    torch_input_dtype = np_to_torch_dtype(input_dtype)
+    torch_output0_dtype = np_to_torch_dtype(output0_dtype)
+    torch_output1_dtype = np_to_torch_dtype(output1_dtype)
+
+    model_name = tu.get_model_name(
+        "python_nobatch" if max_batch == 0 else "python", input_dtype,
+        output0_dtype, output1_dtype)
+    # handle for -1 (when variable) since can't create tensor with shape of [-1]
+    input_shape = [abs(ips) for ips in input_shape]
+    example_input = torch.zeros(input_shape, dtype=torch_input_dtype)
+
+    # Create the model
+    if not swap:
+
+        model = '''
+import numpy as np
+import sys
+from torch import nn
+import torch
+import json
+
+sys.path.append('../../')
+import triton_python_backend_utils as pb_utils
+
+
+class AddSubNet(nn.Module):
+    def __init__(self, output0_dtype, output1_dtype):
+        self.torch_output0_dtype = output0_dtype
+        self.torch_output1_dtype = output1_dtype
+        super(AddSubNet, self).__init__()
+
+    def forward(self, input0, input1):
+        return (input0 + input1).astype(self.torch_output0_dtype), \
+            (input0 - input1).astype(self.torch_output1_dtype)
+
+
+class TritonPythonModel:
+
+    def initialize(self, args):
+        self.model_config = model_config = json.loads(args['model_config'])
+
+        output0_config = pb_utils.get_output_config_by_name(model_config, "OUTPUT0")
+        output1_config = pb_utils.get_output_config_by_name(model_config, "OUTPUT1")
+
+        output0_dtype = pb_utils.triton_string_to_numpy(output0_config['data_type'])
+        output1_dtype = pb_utils.triton_string_to_numpy(output1_config['data_type'])
+        self.model = AddSubNet(output0_dtype, output1_dtype)
+
+    def execute(self, requests):
+        """ This function is called on inference request.
+        """
+        responses = []
+        for request in requests:
+            input_tensors = request.inputs()
+            in_0 = pb_utils.get_input_tensor_by_name(request, "INPUT0")
+            in_1 = pb_utils.get_input_tensor_by_name(request, "INPUT1")
+            if in_0.as_numpy().dtype.type is np.bytes_ or in_0.as_numpy().dtype.type is np.object:
+                out_0, out_1 = self.model(in_0.as_numpy().astype(np.int32), in_1.as_numpy().astype(np.int32))
+                out_tensor_0 = pb_utils.Tensor("OUTPUT0", out_0)
+                out_tensor_1 = pb_utils.Tensor("OUTPUT1", out_1)
+            else:
+                out_0, out_1 = self.model(in_0.as_numpy(), in_1.as_numpy())
+                out_tensor_0 = pb_utils.Tensor("OUTPUT0", out_0)
+                out_tensor_1 = pb_utils.Tensor("OUTPUT1", out_1)
+            responses.append(pb_utils.InferenceResponse([out_tensor_0, out_tensor_1]))
+        return responses
+'''.format(input_shape, torch_input_dtype, example_input, example_input)
```

Comment:
Where are these variables being used in the string placeholder?

Reply:
Fixed